static/code/editor.js

```diff
@@ -130,3 +370,224 @@ Code.Editor.hideButter = function() {
 
 window.addEventListener('load', Code.Editor.init);
 window.addEventListener('message', Code.Editor.receiveMessage, false);
+window.addEventListener('beforeunload', Code.Editor.beforeUnload);
+
+
+Code.Editor.editors = [];
+
+/**
+ * Base class for editors.
+ * @param {string} name User-facing name of editor (used in tab).
+ * @constructor
+ */
+Code.GenericEditor = function(name) {
```

Comment:
Renamed back to GenericEditor.

Reply:
Yeah: not sure I am keen on `GenericEditor` but I will admit it is better than `Editor.Editor`

Maybe `Code.Editor` should be `Code.EditorPane` (or …Frame) and then this could be `Code.Editor`?

Reply:
It's all getting replaced anyway.